### PR TITLE
[1ES Template] Migrate Maestro Build Promotion

### PR DESCRIPTION
**Pull request by epananth**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: eng/common/templates-official/steps/publish-logs.yml [hunk 1]
```diff
@@ -15,35 +14,14 @@ steps:
...
-  condition: always()
-      
-- task: 1ES.PublishBuildArtifacts@1
-  displayName: Publish Logs
-  inputs:
-    PathtoPublish: '$(Build.SourcesDirectory)/PostBuildLogs'
-    PublishLocation: Container
-    ArtifactName: PostBuildLogs
+    filePath: $(Build.SourcesDirectory)/eng/common/post-build/redact-logs.ps1
+    arguments: -InputPath '$(Build.SourcesDirectory)/PostBuildLogs' -BinlogToolVersion ${{parameters.BinlogToolVersion}} -TokensFilePath '$(Build.SourcesDirectory)/eng/BinlogSecretsRedactionFile.txt' '$(publishing-dnceng-devdiv-code-r-build-re)' '$(MaestroAccessToken)' '$(dn-bot-all-orgs-artifact-feeds-rw)' '$(akams-client-id)' '$(akams-client-secret)' '$(microsoft-symbol-server-pat)' '$(symweb-symbol-server-pat)' '$(dn-bot-all-orgs-build-rw-code-rw)' ${{parameters.CustomSensitiveDataList}}
```
Comment (riarenas) [inline eng/common/templates-official/steps/publish-logs.yml:25]: Could we format this again? it's pretty unreadable in a single line.

File: eng/publishing/v3/publish.yml [hunk 2]
```diff
@@ -4,192 +4,132 @@ parameters:
...
...Build.ArtifactStagingDirectory)/PDBArtifacts/' /p:SymbolPublishingExclusionsFile='$(Build.ArtifactStagingDirectory)/ReleaseConfigs/SymbolPublishingExclusionsFile.txt' ${{ parameters.symbolPublishingAdditionalParameters}} /p:MsdlToken=$(microsoft-symbol-server-pat) /p:SymWebToken=$(symweb-symbol-server-pat) /p:BuildQuality='${{ parameters.buildQuality }}' /p:AzdoApiToken='$(dn-bot-all-orgs-build-rw-code-rw)' /p:ArtifactsBasePath='$(Build.ArtifactStagingDirectory)/' /p:BuildId='$(AzDOBuildId)' /p:AzureDevOpsOrg='$(AzDOAccount)' /p:AzureProject='$(AzDOProject)' /p:UseStreamingPublishing='true' /p:StreamingPublishingMaxClients=16 /p:NonStreamingPublishingMaxClients=12 /p:DotNetBuildsPublicUriBase64='$(dotnetbuilds-public-container-uri-base64)' /p:DotNetBuildsPublicChecksumsUriBase64='$(dotnetbuilds-public-container-checksum-uri-base64)' /p:DotNetBuildsInternalUriBase64='$(dotnetbuilds-internal-container-uri-base64)' /p:DotNetBuildsInternalChecksumsUriBase64='$(dotnetbuilds-internal-container-checksum-uri-base64)' /p:DotNetCliMsrcDotNetUriBase64='$(dotnetclimsrc-dotnet-container-uri-base64)' /p:DotNetCliChecksumsMsrcDotNetUriBase64='$(dotnetclichecksumsmsrc-dotnet-container-uri-base64)'
```
Comment (riarenas) [inline eng/publishing/v3/publish.yml:130]: Same about reformatting.

File: eng/common/templates-official/steps/publish-logs.yml (hunk 1 above)
Comment (epananth) [reply eng/common/templates-official/steps/publish-logs.yml:25]: Yes, sorry still working on it!

File: eng/publishing/v3/publish.yml (hunk 2 above)
Comment (epananth) [reply eng/publishing/v3/publish.yml:130]: yep will do :)

**Comment by riarenas** [comment]
Do you perhaps have some test builds to check out?

File: eng/common/templates-official/steps/publish-logs.yml (hunk 1 above)
Comment (riarenas) [reply eng/common/templates-official/steps/publish-logs.yml:25]: aah, I looked at it too early 🙈 I didn't see anything.

**Comment by epananth** [comment]
I think I got this working 
Devdiv https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=9246586&view=results, but dnceng is failing cos of secret rotation. I will work on a test build for it tomo.

**Comment by epananth** [comment]
Dnceng test worked too -> https://dev.azure.com/dnceng/internal/_build/results?buildId=2405026&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=ba23343f-f710-5af9-782d-5bd26b102304

**Comment by epananth** [comment]
Dnceng -https://dev.azure.com/dnceng/internal/_build/results?buildId=2405044&view=logs&s=380ab8c6-4139-55ff-15c4-fb48260f4dca&j=ba23343f-f710-5af9-782d-5bd26b102304

File: eng/common/templates-official/steps/publish-logs.yml [hunk 3]
```diff
@@ -2,48 +2,36 @@ parameters:
   StageLabel: ''
```
Comment (riarenas) [inline eng/common/templates-official/steps/publish-logs.yml:2]: Why are we changing eng/common?

Comment (riarenas) [reply eng/common/templates-official/steps/publish-logs.yml:2]: I think this file looks broken?

**Comment by riarenas** [comment]
We shouldn't be changing the eng/common templates here, just let the log publishing be handled by that template.

File: eng/publishing/v3/publish.yml [hunk 4]
```diff
@@ -4,192 +4,164 @@ parameters:
...
+    - name: MaestroApiEndPoint
+      value: "https://maestro.dot.net"
+    - name: MaestroApiAccessToken
+      value: $(MaestroAccessToken)
+    - name: MaestroApiVersion
+      value: "2020-02-20"
+    - template: /eng/common/templates-official/post-build/common-variables.yml@self
+    templateContext:
+      outputs:
+      - output: pipelineArtifact
```
[the remark this answers is not in the document]
Comment (epananth) [reply eng/publishing/v3/publish.yml:29]: Ahhh, I thought I need to add this and remove the other one. Thank you for pointing it out, will fix this and rerun test.

File: eng/promote-build.yml [hunk 5]
```diff
@@ -1,67 +1,74 @@
...
+    pool:
+      ${{ if eq(variables['System.TeamProject'], 'DevDiv') }}:
+        name: AzurePipelines-EO
+        image: 1ESPT-Windows2019
+        os: windows
+      ${{ else }}:
+        name: NetCore1ESPool-Publishing-Internal
+        image: windows.vs2019.amd64
+        os: windows
+    customBuildTags:
```
Comment (AlitzelMendez) [inline eng/promote-build.yml:66]: I think you can remove this tag :)

Comment (epananth) [reply eng/promote-build.yml:66]: sure will do. :)

File: eng/common/templates-official/steps/publish-logs.yml (hunk 3 above)
Comment (epananth) [reply eng/common/templates-official/steps/publish-logs.yml]: reverted this

**Comment by epananth** [comment]
Devdiv test -https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=9253183&view=results

Dnceng test - https://dev.azure.com/dnceng/internal/_build/results?buildId=2405933&view=results

**Review by riarenas — DISMISSED** [review]
Both the test builds and the YAML look ok to me.

The whitespace removals I can either take or leave if you feel like re-adding the blank lines and you think it adds readability.

**Pull request merged**